python/lsst/daf/butler/formatters/yamlFormatter.py

```diff
@@ -82,7 +117,27 @@ def _writeFile(self, inMemoryDataset, fileDescriptor):
         with open(fileDescriptor.location.path, "w") as fd:
             if hasattr(inMemoryDataset, "_asdict"):
                 inMemoryDataset = inMemoryDataset._asdict()
-            yaml.dump(inMemoryDataset, stream=fd)
+            fd.write(self._toBytes(inMemoryDataset))
+
+    def _toBytes(self, inMemoryDataset):
+        """Write the in memory dataset to a bytestring.
+
+        Parameters
+        ----------
+        inMemoryDataset : `object`
+            Object to serialize
+
+        Returns
+        -------
+        data : `str`
```

Comment:
`bytes` not `str`.

Reply:
Forgot but added. Should be up in minutes.